### PR TITLE
documentation update for `yarn upgrade` command

### PR DESCRIPTION
**Pull request by nsfmc**
Not sure if this is overkill or what, several of the onboarding docs mention this usage of `yarn upgrade` but it wasn't fleshed out in the docs at all, so i decided to add it and experimented with the behavior of the command to give folks a good idea of how it behaves in various cases.

I didn't immediately come across a documentation style guide or anything, but tried to keep things within the house style.

### CODE REVIEW

**Comment by nsfmc** [comment]
sorry, already found an error here, will update pr soon ☹️

**Comment by nsfmc** [comment]
the one change i made to appease the toc url handler was changing the syntax from
```diff
- yarn upgrade <package>
+ yarn upgrade [package]
```
otherwise the slugify-ing process would strip the contents of the <>'ed text

you can see the built version here: https://deploy-preview-195--assembler-pig-23305.netlify.com/en/docs/cli/upgrade

File: en/docs/cli/upgrade.md [hunk 1]
```diff
@@ -30,6 +30,89 @@ success Saved 867 new dependencies.
...
+```diff
+-  "react": "^15.3.2",
++  "react": "latest",
+```
+
 Also see:
 
 - [`yarn add`](./add): add a package to use in your current package.
+- [`yarn tag`](./tag#toc-what-are-tags): What are tags?
+- [Versions of dependencies](../dependency-versions#toc-semantic-versioning): Semantic versioning
```
Comment (thejameskyle) [inline en/docs/cli/upgrade.md]: Can you add all of these to a `additional_reading` field in the jekyll frontmatter? It makes this appear:

screen shot 2016-10-19 at 2 28 35 pm

Comment (nsfmc) [inline en/docs/cli/upgrade.md]: totally, did not know about that at all

**Comment by wyze** [comment]
Looks much better and more detailed than what I put together. Awesome work here @nsfmc!

**Comment by nsfmc** [comment]
i have another commit, slightly larger @thejameskyle but it adds descriptions for subpages as well if they define them, is that overkill here?

**Comment by nsfmc** [comment]
btw, if that last commit goes too far, let me know, i'll revert back

**Comment by nsfmc** [comment]
@thejameskyle any feedback or things i should fixup in this commit?

**Pull request merged**